beam/public/js/scan/scan.js

```diff
@@ -66,18 +83,24 @@ class ScanHandler {
 	}
 	async get_scanned_context(sCode, iQty) {
 		return new Promise(resolve => {
-			const context = this.reduceContext()
-			frappe.xcall('beam.beam.scan.scan', { barcode: sCode, context: context, current_qty: iQty }).then(r => {
-				if (r && r.length) {
-					if (Object.keys(frappe.boot.beam.client).includes(r[0].action)) {
-						let path = frappe.boot.beam.client[r[0].action][0]
-						resolve(path.split('.').reduce((o, i) => o[i], window)(r)) // calls (first) custom built callback registered in hooks
-					} else {
-						resolve(this[String(r[0].action)](r)) // TODO: this only calls the first function
+			if (isLoginPath) {
+				frappe.xcall('scan_login', { barcode: sCode }).then(r => {
```

Comment:
Similar to the `xcall` below, this can just be the following:

```suggestion
				frappe.xcall('beam.beam.scan.user_login.scan_login', { barcode: sCode }).then(r => {
```